frontend/mobilityai/app/src/main/java/com/ai/mobility/mobilityai/MainActivity.java

```diff
@@ -0,0 +1,324 @@
+
+import com.ai.mobility.mobilityai.MetaMotionDeviceAdapter.OnItemClickListener;
+import com.mbientlab.bletoolbox.scanner.BleScannerFragment;
+import com.mbientlab.metawear.Data;
+import com.mbientlab.metawear.MetaWearBoard;
+import com.mbientlab.metawear.Route;
+import com.mbientlab.metawear.Subscriber;
+import com.mbientlab.metawear.android.BtleService;
+import com.mbientlab.metawear.builder.RouteBuilder;
+import com.mbientlab.metawear.builder.RouteComponent;
+import com.mbientlab.metawear.data.Acceleration;
+import com.mbientlab.metawear.data.AngularVelocity;
+import com.mbientlab.metawear.module.Accelerometer;
+import com.mbientlab.metawear.module.GyroBmi160;
+import com.mbientlab.metawear.module.Led;
+import com.mbientlab.metawear.module.Logging;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Random;
+import java.util.UUID;
+
+import bolts.Continuation;
+import bolts.Task;
+
+public class MainActivity extends AppCompatActivity implements ServiceConnection {//, BleScannerFragment.ScannerCommunicationBus {
+    //Metawear classes
+    private BtleService.LocalBinder serviceBinder;
+    private Led led;
+    private MetaWearBoard board;
+    private Accelerometer accelerometer;
+    private GyroBmi160 gyroscope;
+    private Logging logging;
+
+    private MetaMotionService metaMotion;
+
+    private Button led_on, led_off;
+
+    private static final String TAG = "MobilityAI";
+    private static final int REQUEST_ENABLE_BT = 1, PERMISSION_REQUEST_COARSE_LOCATION= 2;
+
+    //RecylerView
+    private RecyclerView m_bleList;
+    private MetaMotionDeviceAdapter m_adapter;
+    private ArrayList<MetaMotionDevice> m_deviceList;
+
+    private ImageButton m_refreshButton;
+
+
+    //Bluetooth Scanning
+    private BluetoothAdapter m_bluetoothAdapter;
+    private BluetoothLeScanner m_bluetoothLeScanner;
+    private boolean m_isScanning = false;
+    private Handler m_handler;
+    private ScanCallback m_leScanCallback;
+    private HashSet<ParcelUuid> m_filterServiceUuids;
+    private BleScannerFragment.ScannerCommunicationBus commBus= null;
+    private boolean m_isScanReady;
+    private ArrayList<BluetoothDevice> m_devices;
+    private static final long SCAN_PERIOD = 10000L;
+
+
+    //TODO: Remove
+    private Random r = new Random();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        initialize();
+
+        setUpBluetoothScanner();
+
+        // Bind the service when the activity is created
+        getApplicationContext().bindService(new Intent(this, BtleService.class),
+                this, Context.BIND_AUTO_CREATE);
+    }
+
+    //TODO: Delete this function once Bluetooth stuff is working
+    private String getRandomName() {
```

Comment:
Is the bluetooth stuff working?

Reply:
Wrong comment, see my new commit b1c33bf for the updated TODO messages.

Reply:
So should we consider adding sample data to the server to get things rolling?